pysat/tests/test_orbits.py

```diff
@@ -128,7 +138,80 @@ def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst
 
-    def test_repeated_orbit_calls_symmetric_single_day_starting_with_last(self):
+    @raises(StopIteration)
+    def test_load_orbits_w_empty_data(self):
+        self.testInst.load(1958, 31)
+        self.testInst.orbits[0]
+        self.testInst.orbits.next()
+
+    def test_less_than_one_orbit_of_data(self):
+        def filter_data(inst):
+            inst.data = inst[0:20]
+        self.testInst.custom.add(filter_data, 'modify')
+        self.testInst.load(2009, 1)
+        self.testInst.orbits.next()
+        # a recusion issue has been observed in this area
+        # checking for date to limit reintroduction potential
+        assert self.testInst.date == pysat.datetime(2009, 1, 1)
+
+    def test_less_than_one_orbit_of_data_two_ways(self):
+        def filter_data(inst):
+            inst.data = inst[0:5]
+        self.testInst.custom.add(filter_data, 'modify')
+        self.testInst.load(2009, 1)
+        # starting from no orbit calls next loads first orbit
+        self.testInst.orbits.next()
+        # store comparison data
+        saved_data = self.testInst.data.copy()
+        self.testInst.load(2009, 1)
+        self.testInst.orbits[0]
+        print('0 ', saved_data.index, self.testInst.index)
+        assert all(self.testInst.data == saved_data)
```

Comment:
In the current test setup, `saved_data` contains data from 01-01-2009, `self.testInst.data` contains data from 12-31-2008.  Not sure what is up here.

Reply:
The orbits iterator goes past day/file boundaries to load complete orbits.